packages/graphiql/src/components/DocExplorer.js

```diff
@@ -54,6 +57,29 @@ export class DocExplorer extends React.Component {
     );
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.schema && 
+      (this.props.schema !== prevProps.schema) && 
+      this.props.initialTypeName) {
+      // The schema has changed, and we should render an initial type.
+
+      const typeMap = this.props.schema.getTypeMap();
+      const fullType = typeMap[this.props.initialTypeName];
+
+      if (fullType && this.props.initialFieldName) {
+        if (fullType.getFields) {
+          const fullField = fullType.getFields()[this.props.initialFieldName];
+          if (fullField) {
+            this.showDoc(fullField);
+          }
+          this.showDoc(fullType.getFields()[this.props.initialFieldName]);
+        }
+      } else if (fullType) { // Make sure that fullType exists.
+        this.showDoc(fullType);
+      }
```

Comment:
If the field doesn't exist but the type does, I think we should display the type, so something more like:

```js
      const { initialTypeName, initialFieldName } = this.props;
      const typeMap = this.props.schema.getTypeMap();
      const fullType = typeMap[initialTypeName];
      const fieldMap = fullType ? fullType.getFields() : null;
      const fullField = fieldMap && initialFieldName ? fieldMap[initialFieldName] : null;

      if (fullField) {
        this.showDoc(fullField);
      } else if (fullType) {
        this.showDoc(fullType);
      }
```